From d4712049be93d0ad3137a9b9518ce0e31273c407 Mon Sep 17 00:00:00 2001
From: Meng Wei <wei.meng@intel.com>
Date: Mon, 25 Mar 2019 10:10:16 +0800
Subject: [PATCH 1114/1214] media: intel-ipu4: Fixing static analysis issue in
 common code

Fixing static analysis coding issue of IPU driver common code

Change-Id: Ib44a23cc0b09d6027b457bd7e66b41a6e781c69e
Tracked-On: PKT-2287
Signed-off-by: mingdaxu <mingda.xu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-dma.c | 6 ++++--
 drivers/media/pci/intel/ipu.c     | 5 ++++-
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-dma.c b/drivers/media/pci/intel/ipu-dma.c
index 24b45a2..290134a 100644
--- a/drivers/media/pci/intel/ipu-dma.c
+++ b/drivers/media/pci/intel/ipu-dma.c
@@ -254,7 +254,8 @@ static void ipu_dma_free(struct device *dev, size_t size, void *vaddr,
 	if (WARN_ON(!area->pages))
 		return;
 
-	WARN_ON(!iova);
+	if (WARN_ON(!iova))
+		return;
 
 	size = PAGE_ALIGN(size);
 
@@ -319,7 +320,8 @@ static void ipu_dma_unmap_sg(struct device *dev,
 	if (!nents)
 		return;
 
-	WARN_ON(!iova);
+	if (WARN_ON(!iova))
+		return;
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 8, 0)
 	if (!dma_get_attr(DMA_ATTR_SKIP_CPU_SYNC, attrs))
diff --git a/drivers/media/pci/intel/ipu.c b/drivers/media/pci/intel/ipu.c
index 04293f4..3f9b82a 100644
--- a/drivers/media/pci/intel/ipu.c
+++ b/drivers/media/pci/intel/ipu.c
@@ -330,12 +330,15 @@ int request_cpd_fw(const struct firmware **firmware_p, const char *name,
 		*firmware_p = fw;
 	} else {
 		tmp = (struct firmware *)kzalloc(sizeof(struct firmware), GFP_KERNEL);
-		if (!tmp)
+		if (!tmp) {
+			release_firmware(fw);
 			return -ENOMEM;
+		}
 		tmp->size = fw->size;
 		tmp->data = vmalloc(fw->size);
 		if (!tmp->data) {
 			kfree(tmp);
+			release_firmware(fw);
 			return -ENOMEM;
 		}
 		memcpy((void *)tmp->data, fw->data, fw->size);
-- 
2.7.4

